File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/library/folders/wrappers/HeaderViewWrapper.java
package com.github.anrimian.musicplayer.ui.library.folders.wrappers;

import androidx.annotation.NonNull;

import com.github.anrimian.musicplayer.databinding.PartialStorageHeaderBinding;
import com.github.anrimian.musicplayer.domain.models.folders.FolderFileSource;

import static android.view.View.GONE;
import static android.view.View.OnClickListener;
import static android.view.View.VISIBLE;

/**
 * Created on 01.11.2017.
 */

public class HeaderViewWrapper {

    private final PartialStorageHeaderBinding viewBinding;

    public HeaderViewWrapper(PartialStorageHeaderBinding viewBinding) {
        this.viewBinding = viewBinding;
    }

    public void bind(@NonNull FolderFileSource folder) {
        viewBinding.tvParentPath.setText(folder.getName());
    }

    public void setOnClickListener(OnClickListener listener) {
        viewBinding.headerClickableItem.setOnClickListener(listener);
    }

    public void setVisible(boolean visible) {
        viewBinding.getRoot().setVisibility(visible? VISIBLE: GONE);
    }

}

File: app/src/main/java/com.github.anrimian.musicplayer/ui/library/folders/wrappers/HeaderViewWrapper.kt
package com.github.anrimian.musicplayer.ui.library.folders.wrappers

import android.view.View
import com.github.anrimian.musicplayer.databinding.PartialStorageHeaderBinding
import com.github.anrimian.musicplayer.domain.models.folders.FolderFileSource

/**
 * Created on 01.11.2017.
 */
class HeaderViewWrapper(private val viewBinding: PartialStorageHeaderBinding) {

    fun bind(folder: FolderFileSource) {
        viewBinding.tvParentPath.text = folder.name
    }

    fun setOnClickListener(listener: View.OnClickListener) {
        viewBinding.headerClickableItem.setOnClickListener(listener)
    }

    fun setVisible(visible: Boolean) {
        viewBinding.root.visibility = if (visible) View.VISIBLE else View.GONE
    }

}